test/e2e/transaction.test.ts

```diff
@@ -29,6 +31,7 @@ describe('transactions', () => {
   }
 
   beforeEach(async () => {
+    server.listen(test.network.mockNodePort)
```

Comment:
As this server is non-stateful and does not change between tests at all, it seems to me that starting/stopping a sever per test is a lot of overhead.

IMO we should only start this server once per test run and have methods in order setup/clear down the data.